File: project-signer-runnable/src/main/kotlin/org/jesperancinha/parser/projectsigner/model/ProjectData.java
package org.jesperancinha.parser.projectsigner.model;

import lombok.Builder;
import org.jesperancinha.parser.markdowner.badges.model.BadgeGroup;
import org.jesperancinha.parser.markdowner.badges.model.BadgeType;

import java.util.Map;

@Builder
public record ProjectData(String title, Map<BadgeType, BadgeGroup> badgeGroupMap) {
}

File: project-signer-runnable/src/main/kotlin/org/jesperancinha/parser/projectsigner/model/ProjectData.kt
package org.jesperancinha.parser.projectsigner.model

import org.jesperancinha.parser.markdowner.badges.model.BadgeGroup
import org.jesperancinha.parser.markdowner.badges.model.BadgeType

data class ProjectData(
    val title: String,
    val directory: String?,
    val badgeGroupMap: Map<BadgeType, BadgeGroup>
) {
    fun getNormalizedTitle() = if(title.startsWith("[![")) title else (directory ?: title)
}